docs/topics/opt-in-requirements.md

```diff
@@ -271,21 +357,56 @@ fun getTime(): Time {}
 ```
 
 Note that for some language elements, an opt-in requirement annotation is not applicable:
-* You cannot annotate a backing field or a getter of a property, just the property itself.
-* You cannot annotate a local variable or a value parameter.
+
+* You can't annotate a backing field or a getter of a property, just the property itself.
+* You can't annotate a local variable or a value parameter.
+
+## Require opt-in to extend API
+
+There may be times when you want more granular control over which specific parts of your API can be used and
+extended. For example, when you have some API that is stable to use but:
+
+* **Unstable to implement** due to ongoing evolution, such as when you have a family of interfaces where you expect to add new abstract functions without default implementations.
+* **Closed to third-party implementations** due to internal or technical reasons, such as an API that you want to be sealed but can't for technical reasons.
+* **Delicate or fragile to implement**, such as individual functions that need to behave in a coordinated manner.
+* **Has a contract that may be weakened in the future** in a backwards-incompatible manner for external implementations, such as changing an input parameter `T` to a nullable version `T?` where the code didn't previously consider `null` values.
+
+In such cases, you want to require users to opt in to your API before they can extend it further.
+For example, by inheriting from the API or declaring implementations for abstract functions. Using the [`@SubclassOptInRequired`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/)
+annotation, you can enforce this behavior for [open](inheritance.md) or [abstract classes](classes.md#abstract-classes) and [non-functional interfaces](interfaces.md).
+
+To add the opt-in requirement to an API element, use the [`@SubclassOptInRequired`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/)
+annotation with a reference to the annotation class:
+
+```kotlin
+@RequiresOptIn(
+ level = RequiresOptIn.Level.WARNING,
+ message = "Interfaces in this library are experimental"
+)
+annotation class UnstableApi()
+
+@SubclassOptInRequired(UnstableApi::class)
+interface CoreLibraryApi // An interface requiring opt-in to extend
+```
+
+Note that when you use the `@SubclassOptInRequired` annotation to require opt-in, the requirement is not propagated to
```

Comment:
Yes. Since the doc is split into two parts - one for those who want to opt in to API and one for those who want to add opt-in requirements for their API. This way if only one part is read, the information isn't missed. It seems relevant for both use cases.